docs/resources/elasticsearch.md.erb

```diff
@@ -0,0 +1,123 @@
+---
+title: The Elasticsearch Resource
+---
+
+# elasticsearch
+
+Use the Elasticsearch InSpec audit resource to test the Elasticsearch configuration
+settings. The data is extracted from the live elasticsearch cluster using the curl
+command `curl 'ES_IP':'ES_PORT'/_nodes/`. The resource makes uses of FilterTable and
+its functionality can be used.
+
+## Syntax
+
+An `elasticsearch` resource is called with the elasticsearch IP and port;
+Default IP:localhost; Port:9200
+
+    describe elasticsearch(ES_IP,ES_PORT) do
+      its('property') { should eq value }
+    end
+
+where
+
+* `.where()` filter may specify a specific item and value, to which the matchers are compared
+* `.entries()` filter will return the entries that satisfy the where() condition
+* `'cluster_name'`, `'node_id'`, and `'node_name'` etc. are valid filters for this resource
+* `'property'` is a configuration setting to be tested
+* `{ should eq 'value' }` is the value that is expected
+
+## Supported Properties
+    * `'cluster_name'`, `'node_id'`, `'node_name'`,`'host'`, `'ip'`, `'version'`, `'build_hash'`
+    * `'total_indexing_buffer'`, roles'`, `'conf_path'`, `'data_path'`, `'log_path'`, `'home_path'`
+    * `'client_type'`, `'http_type'`, `'transport_type'`, `'os'`, `'process'`, `'mlockall'`
+    * `'transport'`, `'http'`, `'pidfile'`, `'transport_address'`, `'pid'`, `'jvm_version'`,
+    * `'vm_name'`, `'vm_version'`, `'vm_vendor'`, `'start_time_in_millis'`, `'mem'`
+    * `'gc_collectors'`, `'memory_pools'`, `'input_arguments'`, `'module_list'`,
+    * `'using_compressed_ordinary_object_pointers'`
+
+## Property Examples and Return Types
+
+### cluster_name ([String])
+
+    describe elasticsearch.where { node_name == 'package-centos-72' } do
+      its('cluster_name.first') { should match 'elasticsearch' }
```

Comment:
Since the `cmp` matcher will change a single-element array to a non-array, I'd change this to:

```
its('cluster_name') { should cmp 'elasticsearch' }
```